### PR TITLE
Portal aad cleanup

### PR DESCRIPTION
**Pull request by facchettos**
### Which issue this PR addresses:
depends on #2740 , blocked by our integration setup 

moves the route definition to portal where it is supposed to be. Also fixes the AAD middleware which uses its own way of storing the user info instead of getting it from the jwt we are getting from AAD

Fixes

### CODE REVIEW

File: pkg/portal/middleware/aad.go
Comment (facchettos) [reply pkg/portal/middleware/aad.go:173]: state is generated with a uuid generator so it's not constant

File: pkg/portal/security_test.go
Comment (facchettos) [inline pkg/portal/security_test.go:1]: This test's only value was to ensure we are running 2 middlewares (Log and AAD). with the redesign and various fixes it is hard to mock it in the same way. Since we are moving to chi in the next days and that chi allows us to list the middlewares applied on routes, I removed it to avoid unnecessary work and will try to merge the PRs at the same time so we don't lose those tests

File: pkg/portal/middleware/aad.go
Comment (yehornaumenko) [inline pkg/portal/middleware/aad.go:259]: why not return the entire claim object?

**Review by yehornaumenko — CHANGES_REQUESTED** [review]
Now it looks much better, good job

File: pkg/portal/middleware/aad.go
Comment (facchettos) [reply pkg/portal/middleware/aad.go:278]: if it's not allowed, we might want to log things, so we have a trace (e.g. in case of malicious attempts of pen testing). for the other you are right, I changed it

File: pkg/portal/middleware/aad.go
Comment (facchettos) [reply pkg/portal/middleware/aad.go:270]: I ended up copy-pasting a little bit too much, good catch :D

File: pkg/portal/middleware/aad.go
Comment (facchettos) [reply pkg/portal/middleware/aad.go:259]: it just avoids having to deal with a claim later on, because after that we really only need about the username and the groups, not the claim.

**Review by yehornaumenko — APPROVED** [review]

**Review by petrkotas — APPROVED** [review]
I think this is a good upgrade. We can merge this, when we are doing next release, because this is going to be fully tested in canary as we do not have AAD tenant setup and prepared in CI and INT.

**Review by petrkotas — APPROVED** [review]
Thanks @facchettos great work. Lets get it in and test in canary

File: pkg/portal/portal.go
Comment (hawkowl) [inline pkg/portal/portal.go:166]: Since we will only ever test this in INT (since we don't/can't run the portal e2e in PROD), can we make it so that we need to give the flag AND be in local development mode/CI for this to work? So, I guess, `disableOauthOption == "true" && (env.IsCI() || env.IsLocalDevelopmentMode())`?

File: pkg/portal/portal.go
Comment (hawkowl) [inline pkg/portal/portal.go:170]: I don't think we need to log this when we're starting in regular mode, especially as error. I would just remove it.

File: pkg/portal/ssh/ssh.go
Comment (hawkowl) [inline pkg/portal/ssh/ssh.go:138]: We need a return here, I think?

File: test/e2e/setup.go
Comment (hawkowl) [inline test/e2e/setup.go:157]: Is this const used?

File: pkg/deploy/devconfig.go
Comment (hawkowl) [inline pkg/deploy/devconfig.go:104]: Do we need to disable oauth for dev use? I think we have the AAD groups set up for it already, it's just CI that's a problem.

**Review by hawkowl — CHANGES_REQUESTED** [review]
Minor comments, overall looks good. The only thing I'd want to confirm before merging is the disable flag checking logic and whether that missing return is actually needed, the rest are optional.

File: pkg/portal/portal.go
Comment (facchettos) [reply pkg/portal/portal.go:166]: pr e2e is actually running with local development mode, I had that condition before but it wasn't working. Something like `(disableOauthOption == "true" && env.IsCI()) || env.IsLocalDevelopmentMode()` should work though

File: pkg/portal/portal.go
Comment (facchettos) [reply pkg/portal/portal.go:170]: this is some leftover from debugging

File: test/e2e/setup.go
Comment (facchettos) [reply test/e2e/setup.go:157]: yeah no, also some leftover from another try

File: pkg/deploy/devconfig.go
Comment (facchettos) [reply pkg/deploy/devconfig.go:104]: ok will try that

**Pull request closed without merging**
